index.js

```diff
@@ -211,10 +211,21 @@ ObjectTemplate.create = function create(name, properties) {
  * @returns {*} the object template
  */
 ObjectTemplate.extend = function extend(parentTemplate, name, properties) {
+    var props;
+    var createProps;
+
     if (!parentTemplate.__objectTemplate__) {
         throw new Error('incorrect parent template');
     }
 
+    if (typeof(name) != 'undefined' && name.name) {
```

Comment:
We can do that was just following the pattern in `objectTemplate.create`. Should I make the change there as well?